src/mavedb/routers/authorization.py

```diff
@@ -0,0 +1,73 @@
+import logging
+from enum import Enum
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+from typing import Union, Optional
+
+from mavedb import deps
+from mavedb.lib.authentication import get_current_user, UserData
+from mavedb.lib.permissions import has_permission, Action
+from mavedb.lib.logging import LoggedRoute
+from mavedb.lib.logging.context import logging_context, save_to_logging_context
+from mavedb.models.experiment import Experiment
+from mavedb.models.experiment_set import ExperimentSet
+from mavedb.models.score_set import ScoreSet
+
+router = APIRouter(
+    prefix="/api/v1",
+    tags=["authorizations"],
+    responses={404: {"description": "Not found"}},
+    route_class=LoggedRoute,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ModelName(str, Enum):
+    experiment = "experiment"
+    experiment_set = "experiment-set"
+    score_set = "score-set"
+
+
+@router.get(
+    "/user-is-authorized/{model_name}/{urn}/{action}",
```

Comment:
Love how generic this is. This was mostly on me because of how I suggested the name of the function, but I think it'd make more sense if we put this in a file named `permissions.py` and made this route `/user-is-permitted/{model_name}/{urn}/{action}`.

 The word authorization has a distinct meaning already (sign the user in) and this is more about checking a users' permissions.

Reply:
Your previous suggestions gave me a hint for this generic function. I love this one too. I agree permission is better.